src/processing/BoardState.java

```diff
@@ -0,0 +1,211 @@
+			if (queue_at_pos == null) {
+				continue;
+			}
+			PriorityQueueObject top_obj = (PriorityQueueObject) queue_at_pos.peek();
+			if (top_obj == null) {
+				continue;
+			}
+
+			Integer curr = count.get(top_obj.objectId);
+			if (curr == null) {
+				curr = 0;
+			}
+			count.put(top_obj.objectId, curr + 1);
+		}
+
+		Integer maxOccurrence = 0;
+		ObjectId maxOccObjectId = null;
+		Iterator <Map.Entry<ObjectId, Integer>> it = count.entrySet().iterator();
+
+		while (it.hasNext()) {
+			Map.Entry <ObjectId, Integer> mapElement = (Map.Entry <ObjectId, Integer>) it.next();
+			if (maxOccurrence < mapElement.getValue()) {
+				maxOccurrence = mapElement.getValue();
+				maxOccObjectId = mapElement.getKey();
+			}
+		}
+
+		return maxOccObjectId;
+	}
+
+	/**
+	 * For each Position (x, y) in the input ArrayList, insert the PriorityQueueObject into the
+	 * Priority Queue at the position (x, y) present in the map
+	 */
+	private synchronized void insertIntoPQ (ArrayList <Position> positions, PriorityQueueObject toBeInserted) {
+		for (Position pos : positions) {
+			PriorityQueue <PriorityQueueObject> pq = posToObjects.get(pos);
+			if (pq == null) {
+				posToObjects.put(pos, new PriorityQueue<PriorityQueueObject>(20, new ObjectComparator()));
+			}
+			posToObjects.get(pos).add(toBeInserted);
+		}
+	}
+
+	/**
+	 * For each Position (x, y) in the input ArrayList, delete the PriorityQueueObject from the
+	 * Priority Queue at the position (x, y) present in the map, if it exists in the Priority Queue
+	 */
+	private synchronized void removeFromPQ (ArrayList <Position> positions, PriorityQueueObject objectToBeRemoved) {
+		for (Position pos : positions) {
+			PriorityQueue <PriorityQueueObject> pq = posToObjects.get(pos);
+			if (pq == null) {
+				continue;
+			}
+			pq.remove(objectToBeRemoved);
+		}
+	}
+
+	/**
+	 * Read the ObjectId field from the BoardObject and
+	 * insert the (ObjectId, BoardObject) key, value pair in the map
+	 */
+	private synchronized void insertObjectIntoMap(BoardObject boardObject) {
+		ObjectId objId = boardObject.getObjectId();
+		objIdToBoardObject.put(objId, boardObject);
+	}
+
+	/**
+	 * Remove the input ObjectId's BoardObject from the map
+	 * Thus, the key, value pair (ObjectId, BoardObject) gets deleted from the map
+	 */
+	private synchronized BoardObject removeObjectFromMap(ObjectId objId) {
+		return objIdToBoardObject.remove(objId);
+	}
+
+
+	/* APIs to Update maps to be used internally by other Processing Module classes */
+
+	/**
+	 * Reads the ObjectId, Timestamp, ArrayList<Position> fields of the input BoardObject
+	 * calls insertIntoPQ and insertObjectIntoMap to update both the maps
+	 */
+	public void insertObjectIntoMaps (BoardObject boardObject) {
+		ObjectId objId = boardObject.getObjectId();
+		Timestamp timestamp = boardObject.getTimestamp();
+		ArrayList<Pixel> pixels = boardObject.getPixels();
+
+		ArrayList<Position> positions = new ArrayList<Position>();
```

Comment:
@wildcat26 @himanshu2047 @sakshi1603 
I guess this piece of code is being used used quite often, should I put it in my board object class with public access ?